echopype/tests/visualize/test_plot.py

```diff
@@ -175,3 +177,142 @@ def test_plot_mvbs(
+    ],
+)
+def test_water_level_echodata(water_level, expect_warning):
+    from echopype.echodata import EchoData
+    from echopype.visualize.api import _add_water_level
+
+    filepath = ek60_path / "ncei-wcsd" / "Summer2017-D20170719-T211347.raw"
+    sonar_model = "EK60"
+    range_kwargs = {}
+
+    echodata = echopype.open_raw(
+        sonar_model=sonar_model, raw_file=filepath, xml_path=None
+    )
+
+    range_in_meter = echodata.compute_range(
+        env_params=range_kwargs.get('env_params', {}),
+        azfp_cal_type=range_kwargs.get('azfp_cal_type', None),
+        ek_waveform_mode=range_kwargs.get('waveform_mode', 'CW'),
+        ek_encode_mode=range_kwargs.get('encode_mode', 'power'),
+    )
+    single_array = range_in_meter.isel(frequency=0, ping_time=0).values
+    no_water_level = False
+    if isinstance(water_level, list):
+        water_level = water_level[0]
+        echodata.platform = echodata.platform.drop_vars('water_level')
+        no_water_level = True
+
+    if isinstance(water_level, xr.DataArray):
+        if 'frequency' in water_level.dims:
+            original_array = single_array + water_level.isel(frequency=0).values
+    elif isinstance(water_level, bool) and water_level is True:
+        if no_water_level is False:
+            original_array = (
+                single_array
+                + echodata.platform.water_level.isel(frequency=0, ping_time=0).values
+            )
+        else:
+            original_array = single_array
+    elif isinstance(water_level, (int, float)):
+        original_array = single_array + water_level
+    else:
+        original_array = single_array
+
+    results = None
+    try:
+        if expect_warning:
+            with pytest.warns(UserWarning):
+                results = _add_water_level(
+                    range_in_meter=range_in_meter,
+                    water_level=water_level,
+                    data_type=EchoData,
+                    platform_data=echodata.platform,
+                )
+        else:
+            results = _add_water_level(
+                range_in_meter=range_in_meter,
+                water_level=water_level,
+                data_type=EchoData,
+                platform_data=echodata.platform,
+            )
+    except Exception as e:
+        assert isinstance(e, ValueError)
+        assert str(e) == 'Water level must have any of these dimensions: frequency, ping_time, range_bin'  # noqa
+
+    if isinstance(results, xr.Dataset):
+        final_array = results.isel(frequency=0, ping_time=0).values
+
+        assert np.array_equal(original_array, final_array)
+
+
+@pytest.mark.parametrize(
+    ("water_level", "expect_warning"),
+    [
+        (True, True),
+        (False, True),
+        (xr.DataArray(np.array(50.0)).expand_dims({'frequency': 3}), False),
+        (xr.DataArray(np.array(50.0)), False),
+        (10, False),
+        (30.5, False),
+    ],
+)
+def test_water_level_dataset(water_level, expect_warning):
```

Comment:
suggest changing this to `test_water_level_Sv_dataset` to specify what the dataset is.

Reply:
```suggestion
def test_water_level_Sv_dataset(water_level, expect_warning):
```